ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -382,58 +392,162 @@ azure_object_size(
+        return EINVAL;
+    }
+
+    // Check if object exists.
+    bool exists;
+    int ret;
+    size_t size = 0;
+    if ((ret = azure_fs->azure_conn->object_exists(std::string(name), exists, size)) != 0) {
+        std::cerr << "azure_file_open: object_exists request to Azure failed." << std::endl;
+        return ret;
+    }
+    if (exists == false) {
+        std::cerr << "azure_file_open: no such file named " << name << "." << std::endl;
+        return EINVAL;
+    }
+
+    // Check if there is already an existing file handle open.
+    auto fh_iterator = std::find_if(azure_fs->azure_fh.begin(), azure_fs->azure_fh.end(),
+      [name](azure_file_handle *fh) { return fh->name.compare(name) == 0; });
+
+    // Active file handle for file exists, increment reference count
+    if (fh_iterator != azure_fs->azure_fh.end()) {
+        (*fh_iterator)->reference_count++;
+        *file_handlep = reinterpret_cast<WT_FILE_HANDLE *>(*fh_iterator);
+        return 0;
+    }
+
+    // No active file handle, create a new file handle.
+    azure_file_handle *azure_fh;
+    try {
+        azure_fh = new azure_file_handle;
+    } catch (std::bad_alloc &e) {
+        std::cerr << std::string("azure_file_open: ") + e.what() << std::endl;
+        return ENOMEM;
+    }
+    azure_fh->name = name;
+    azure_fh->reference_count = 1;
+    azure_fh->fs = azure_fs;
+
+    // Define functions needed for Azure with read-only privilleges.
+    azure_fh->fh.close = azure_file_close;
+    azure_fh->fh.fh_advise = nullptr;
+    azure_fh->fh.fh_extend = nullptr;
+    azure_fh->fh.fh_extend_nolock = nullptr;
+    azure_fh->fh.fh_lock = azure_file_lock;
+    azure_fh->fh.fh_map = nullptr;
+    azure_fh->fh.fh_map_discard = nullptr;
+    azure_fh->fh.fh_unmap = nullptr;
+    azure_fh->fh.fh_read = azure_file_read;
+    azure_fh->fh.fh_size = azure_file_size;
+    azure_fh->fh.fh_sync = nullptr;
+    azure_fh->fh.fh_sync_nowait = nullptr;
+    azure_fh->fh.fh_truncate = nullptr;
+    azure_fh->fh.fh_write = nullptr;
+    azure_fh->fh.name = strdup(name);
+
+    // Exclusive Access is required when adding file handles to list of file handles.
+    // lock_guard will unlock automatically when the scope is exited.
+    {
+        std::lock_guard<std::mutex> lock_guard(azure_fs->fh_mutex);
+        azure_fs->azure_fh.push_back(azure_fh);
+    }
+    *file_handlep = &azure_fh->fh;
+
     WT_UNUSED(session);
-    WT_UNUSED(name);
-    WT_UNUSED(file_type);
-    WT_UNUSED(flags);
-    WT_UNUSED(file_handlep);
+    WT_UNUSED(size);
 
     return 0;
 }
 
+// File handle close.
 static int
 azure_file_close(WT_FILE_HANDLE *file_handle, WT_SESSION *session)
 {
-    WT_UNUSED(file_handle);
+    azure_file_handle *azure_fh = reinterpret_cast<azure_file_handle *>(file_handle);
+    // If there are other active instances of the file being open, do not close file handle.
+    if (--azure_fh->reference_count != 0) {
```

Comment:
we don't need braces for one line if statements

Reply:
fixed :)